### PR TITLE
Add CI workflow for building and testing

### LINKED ISSUE #26: No CI for building and testing
There should be CI for building and testing the project so that every pull request automatically goes through additional verification and passes all tests.

### PR DESCRIPTION
**Pull request by nhatdongdang**
## Overview
This pull request adds a new CI workflow for building and testing the mcpp library.
## Changes Made
- Added a new GitHub Actions workflow file for CI.
- Configured the workflow to run on `ubuntu-latest`.
- Included steps for setting up Java, cloning repositories, starting the Spigot server, building the project, and running tests.
## Notes
- Won't work for test cases that depend on the player joining the server
## Linked issue 
#26

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            - name: Build file
+              run: cmake -Bbuild
+
+            - name: Run  local_tests
+              working-directory: ./build
+              run: make local_tests && ./test/local_tests
+
+            - name: Run test_suite
+              working-directory: ./build
+              run: make test_suite && ./test/test_suite
```
Comment (rozukke) [inline .github/workflows/ci.yml]: `test_suite` already includes local tests, and this should probably be run using `ctest` unless there is any issue with that.

**Review by rozukke — CHANGES_REQUESTED** [review]
Thanks for the setup, just a few tweaks and it should be good to go.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (s4059881) [reply .github/workflows/ci.yml]: Thank you for your feedback. I have already fixed the issue, and the test suite is now set up to run using ctest. Please let me know if you encounter any further issues.

Comment (nhatdongdang) [reply .github/workflows/ci.yml]: Thank you for your feedback. I have already fixed the issue, and the test suite is now set up to run using ctest. Please let me know if you encounter any further issues.

**Review by rozukke — APPROVED** [review]
Cheers, LGTM 👍

**Pull request merged**